search2/python/search/index/listener.py

```diff
@@ -190,15 +26,53 @@ def main():
     )
     args = parser.parse_args()
 
-    config_path_base_dir = args.conf_dir
+    conf_dir = args.conf_dir
     with open(args.queue_conf, "r", encoding="utf-8") as queue_config_file:
         queue_conf = YAML(typ="safe").load(queue_config_file)
 
     with open(args.search_conf, "r", encoding="utf-8") as search_config_file:
         search_conf = YAML(typ="safe").load(search_config_file)
 
-    listen(queue_conf, search_conf, config_path_base_dir)
+    def handler_fn(publisher: Publisher, job_details: dict):
+        m_path = get_config_file_path(conf_dir, job_details["assembly"], '.mapping.y*ml')
+
+        with open(m_path, 'r', encoding='utf-8') as f:
+            mapping_conf = YAML(typ="safe").load(f)
+
+        tar_path: Optional[str] = None
+        annotation_path: Optional[str] = None
+        input_file_names = job_details['inputFileNames']
+
+        if input_file_names.get('archived'):
+            tar_path = os.path.join(
+                job_details['inputDir'], job_details['inputFileNames']['archived'])
+        else:
+            annotation_path = os.path.join(job_details['inputDir'],
+                                           job_details['inputFileNames']['annotation'])
+
+        return go(index_name=job_details["indexName"],
+                  tar_path=tar_path,
+                  annotation_path=annotation_path,
+                  mapping_conf=mapping_conf,
+                  search_conf=search_conf,
+                  publisher=publisher)
+
+    def submit_msg_fn(base_msg: dict, job_details: dict): # pylint: disable=unused-argument
```

Comment:
why don't we use job_details here?  In any case you can suppress the warning by renaming as `_job_details`, which also makes it clearer  to the reader that you're ignoring the arg.